feat: Add React support

mstallmo opened the pull request:
Made a couple of minor changes to import resolution and development HTML generation to support React apps.

**Import Resolution**
When Vite builds the React files it does not transform the key entries from `jsx` to `js`.  Previously in-vite was missing entires for `jsx` files when resolving imports causing the js code to not be loaded by the page.

**Development HTML**
When running a React app in development Vite expects there to be a React preamble module loaded and reports an error in the browser if it is missing. This change checks to see if there is a file ending in `.jsx` present in the entrypoints and inserts the React preamble if a `.jsx` file is found.

HiImJulien commented:
This looks solid. I am just worried, what would happen if a different framework, that supports JSX (for instance: SolidJS) is being used.

Since this library is inspired by Laravel's Vite integration, I think we should consider creating a separate directive such as [`viteReactRefresh`](https://laravel.com/docs/11.x/vite#react) or an additional parameter (e.g. `vite(resources=[...], react=True)`). 

Would you mind moving it to a separate function that generates the strings? I'll take care of integrating it with the templating engines, if you'd like.

mstallmo commented:
Yeah I'm happy to move the react refresh stuff out into another function! I'll work on this today.

mstallmo commented:
@HiImJulien just updated the react refresh implementation into it's own separate directive. Also implemented for both supported integrations and updated some tests

HiImJulien commented:
I've merged your changes. I'll have a closer look tomorrow (~ 4 PM CEST) and push a new version!
